cloud/amazon/ec2_asg.py

```diff
@@ -503,33 +506,24 @@ def create_autoscaling_group(connection, module):
             module.fail_json(msg=str(e))
         return(changed, asg_properties)
 
-
 def delete_autoscaling_group(connection, module):
     group_name = module.params.get('name')
     groups = connection.get_all_groups(names=[group_name])
     if groups:
         group = groups[0]
-        group.max_size = 0
-        group.min_size = 0
-        group.desired_capacity = 0
-        group.update()
-        instances = True
-        while instances:
-            tmp_groups = connection.get_all_groups(names=[group_name])
-            if tmp_groups:
-                tmp_group = tmp_groups[0]
-                if not tmp_group.instances:
-                    instances = False
-            time.sleep(10)
-
-        group.delete()
-        while len(connection.get_all_groups(names=[group_name])):
-            time.sleep(5)
-        changed=True
-        return changed
+        group.shutdown_instances()
```

Comment:
I'd be worried that this wouldn't handle an instance that's launching but isn't in a runnable state. PR submitter said this works on IRC, but I'd personally want to see it to believe it.